src/commands/gen-api-models/parse.ts

```diff
@@ -3,17 +3,119 @@
  */
 
 import { ITuple2, Tuple2, Tuple3 } from "@pagopa/ts-commons/lib/tuples";
-import { OpenAPIV2 } from "openapi-types";
+import { OpenAPIV2, IJsonSchema } from "openapi-types";
 import { uncapitalize } from "../../lib/utils";
 import {
   ExtendedOpenAPIV2SecuritySchemeApiKey,
   IAuthHeaderParameterInfo,
+  IDefinition,
   IHeaderParameterInfo,
   IOperationInfo,
   IParameterInfo,
   ISpecMetaInfo,
   SupportedMethod
 } from "./types";
+
+// eslint-disable-next-line prefer-arrow/prefer-arrow-functions
+function parseInnerDefinition(source: IJsonSchema): IDefinition {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const looselySource = source as any;
+
+  if (looselySource.$ref) {
+    return looselySource;
+  }
+
+  // OAS2 does not support disjointed unions with oneOf,
+  //  so we introduced "x-one-of" custom field in association with allOf
+  const { oneOf, allOf } =
+    typeof looselySource.oneOf === "undefined" &&
+    "x-one-of" in looselySource &&
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    looselySource["x-one-of"]
+      ? { allOf: undefined, oneOf: looselySource.allOf }
+      : {
+          allOf: looselySource.allOf,
+          oneOf: looselySource.oneOf
+        };
+
+  // enum used to be defined with "x-extensible-enum" custom field
+  const enumm = looselySource.enum
+    ? looselySource.enum
+    : "x-extensible-enum" in looselySource
+    ? // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      looselySource["x-extensible-enum"]
```

Comment:
there's no need of it

```suggestion
    ? looselySource["x-extensible-enum"]
```